molo/core/content_import/helpers/parse.py

```diff
@@ -0,0 +1,50 @@
+from molo.core.content_import.helpers.locales import locales_not_in_repo
+
+
+def parse_validate_content(repos, locales):
+    main, errors = get_main(locales)
+    errors = errors + check_languages_not_in_repos(repos, locales)
+
+    return {
+        'main': main,
+        'children': get_children(main, locales),
+        'errors': errors
+    }
+
+
+def check_languages_not_in_repos(repos, locales):
+    errors = [check_languages_not_in_repo(repo, locales) for repo in repos]
+    return [error for error in errors if error is not None]
+
+
+def check_languages_not_in_repo(repo, locales):
+    missing = locales_not_in_repo(repo, locales)
+    if missing:
+        return {
+            'type': 'languages_not_in_repo',
+            'details': {
+                'repo': repo.name,
+                'locales': missing
+            }
+        }
+    else:
+        return None
+
+
+def get_children(main, locales):
```

Comment:
probably needs a better name (`get_children_locales` perhaps?), but :bikeshed:, doesn't need to be changed in this PR